src/vs/editor/contrib/gotoSymbol/browser/link/goToDefinitionAtPosition.ts

```diff
@@ -117,6 +119,14 @@ export class GotoDefinitionAtPositionEditorContribution implements IEditorContri
 			return;
 		}
 
+		// check if mouse event is on an inlay hint
+		if (mouseEvent.target.type === MouseTargetType.CONTENT_TEXT) {
```

Comment:
This brings in a dependency between inlay hints and gotoSymbol and that's not really wanted. I'd say it's sufficient to check on L116 that mouse if over `CONTEXT_TEXT`, e.g if it's not don't start definition from mouse

Reply:
I don't think that works, since both normal text and inlay hint has `CONTEXT_TEXT`. But we probably can remove the `options?.attachedData instanceof RenderedInlayHintLabelPart` part and disable definition for all forms of `ModelDecorationInjectedTextOptions`. Would it be fine in this way?

Reply:
I don't think that works, since both normal text and inlay hint has `CONTEXT_TEXT`. But we probably can remove the `options?.attachedData instanceof RenderedInlayHintLabelPart` part and disable definition for all forms of `ModelDecorationInjectedTextOptions`. Would it be fine in this way?